lib/msal-core/src/ScopeSet.ts

```diff
@@ -141,15 +130,49 @@ export class ScopeSet {
      * Appends extraScopesToConsent if passed
      * @param {@link AuthenticationParameters}
      */
-    static appendScopes(reqScopes: Array<string>, reqExtraScopesToConsent: Array<string>): Array<string> {
-        if (reqScopes) {
-            const convertedExtraScopes = reqExtraScopesToConsent ? this.trimAndConvertArrayToLowerCase([...reqExtraScopesToConsent]) : null;
+    static appendScopes(reqScopes: Array<string>, scopesToAppend: Array<string>): Array<string> {
+        if(reqScopes) {
+            const convertedScopesToAppend = scopesToAppend ? this.trimAndConvertArrayToLowerCase([...scopesToAppend]) : null;
             const convertedReqScopes = this.trimAndConvertArrayToLowerCase([...reqScopes]);
-            return convertedExtraScopes ? [...convertedReqScopes, ...convertedExtraScopes] : convertedReqScopes;
+            return convertedScopesToAppend ? [...convertedReqScopes, ...convertedScopesToAppend] : convertedReqScopes;
         }
         return null;
```

Comment:
Would it be better to never set `convertedScopesToAppend` to `null` and just leave it empty, and so then we just return the first expression of the ternary? This avoids playing around with `null` more than we need to.

Reply:
@jmckennon if I understood correctly, what you are suggesting is:

```js
    static appendScopes(reqScopes: Array<string>, scopesToAppend: Array<string>): Array<string> {
        if(reqScopes) {
            const convertedReqScopes = this.trimAndConvertArrayToLowerCase([...reqScopes]);
            return [...convertedReqScopes, ...scopesToAppend];
        }
        return null;
    }
```

Assuming my understanding is correct, I see a couple of problems:

1. `scopesToAppend` need to be trimmed and downcased before being appended, which would be skipped in this example
2. Calling the spread operator on null (scopesToAppend) would create an error.

But I feel like I might have misunderstood your suggestion, could you elaborate a bit please?

Reply:
This is what I meant with my suggestion:
```typescript
    static appendScopes(reqScopes: Array<string>, scopesToAppend: Array<string>): Array<string> {
        if(reqScopes) {
            const convertedScopesToAppend = this.trimAndConvertArrayToLowerCase([...scopesToAppend]);
            const convertedReqScopes = this.trimAndConvertArrayToLowerCase([...reqScopes]);
            return [...convertedReqScopes, ...scopesToAppend];
        }
        return null;
    }
```
Though as you mentioned, if reqScopes is null we'd get an error with the spread operator. I didn't see that previously.

What you've got is fine, approving!

Reply:
Thanks!